app.ts

```diff
@@ -0,0 +1,168 @@
+/** @format */
+
+class DataBase implements DataAggregator {
+    private static instance: DataBase;
+    private users: any[] = [];
+
+    public static getInstance(): DataBase {
+        if (!DataBase.instance) {
+            DataBase.instance = new DataBase();
+        }
+
+        return DataBase.instance;
+    }
+
+    public addUser(user: object): void {
+        this.users.push(user);
+    }
+
+    public getUsers(): object {
+        return this.users;
+    }
+
+    public removeUser(value: string, key: string): object {
+        this.users.filter(item => item[key] !== value)
+        return this.users;
+    }
+
+    public createIterator(): DataBaseIterator {
+        return new DataBaseIterator(this)
+    }
+}
+
+interface DataAggregator {
+    addUser(user: object): void;
+    getUsers(): object;
+    removeUser(value: string, key: string): object;
+    createIterator(): DataBaseIterator;
+}
+
+class DataBaseIterator {
```

Comment:
I don't quite understand how would you use that Iterator to iterate through a whole collection. Please add an example of console.log firstName of every user in the collection using this Iterator